distributed/spans.py

```diff
@@ -364,26 +366,25 @@ def _active_timeseries(self) -> Iterator[tuple[float, bool]]:
         """If this span is the output of :meth:`merge`, yield
         (timestamp, True if at least one input span is active), forward-fill.
         """
-        now = time()
         if self.id != "(merged)":
             yield self.enqueued, True
-            yield self.stop if self.done else now, False
+            yield self.stop, False
             return
 
         events = []
         for child in self.children:
-            events.append((child.enqueued, 1))
-            events.append((child.stop if child.done else now, -1))
+            events += [(child.enqueued, 1), (child.stop, -1)]
         events.sort()
 
         n_active = 0
         for t, delta in events:
-            if not n_active:
-                assert delta > 0
+            # Note: in case of identical timestamps, there may be loops after sorting
+            # were n_active < 0
```

Comment:
IIUC, `child.enqueued <= child.stop`. How about sorting by timestamp only `events.sort(key=lambda event: event[0])`? This should maintain relative ordering and avoid `n_active < 0`. (It took me a moment to understand why `n_active` would ever drop below 0 here.)

Reply:
Done